database/factories/AppointmentFactory.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+use Faker\Generator as Faker;
+
+$factory->define(App\Appointment::class, function (Faker $faker) {
+    if (!App\Clinic::count()) {
```

Comment:
Em không cần phải check có data hay không rồi tạo thêm. Dùng cách tạo data relationship của factory theo link anh gửi là được: https://laravel.com/docs/5.6/database-testing#relationships
Ví dụ:
```php
    return [
        'description' => $faker->text,
        'clinic_id' => function() {
            return factory(App\Clinic::class)->create()->id;
        }
        'user_id' => function() {
            return factory(App\User::class)->create()->id;
        }
        'book_time' => $faker->dateTimeBetween($startDate = '+0 days', $endDate = '+1 months'),
        'status' => $faker->numberBetween($min = 0, $max = 3)
    ];
```

Với cách này thì khi em chạy ở `DatabaseSeeder.php` việc tạo data User, Clinic trước khi tạo Appointment sẽ dẫn đến việc những User, Clinic trước đó không thể có Appointment. Vì ở mỗi Appointment sẽ tạo 1 User, Clinic mới. Tuy nhiên điều này không quan trọng. Vì trong toàn bộ database sẽ có những User/Clinic không có Appointment điều này là hợp lý.